src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -494,105 +464,103 @@ namespace SurfaceGeometry {
+                print(state.files.debug, "{},{:.2R},{:.2R}\n", thisZone.Name, thisZone.ExtGrossWallArea, thisZone.ExtWindowArea);
+            }
             for (int spaceNum : thisZone.spaceIndexes) {
                 auto &thisSpace = state.dataHeatBal->space(spaceNum);
-                CeilCount = 0.0;
-                FloorCount = 0.0;
-                Count = 0;
-                AverageHeight = 0.0;
-                ZCeilAvg = 0.0;
-                ZFlrAvg = 0.0;
-                ZMax = -99999.0;
-                ZMin = 99999.0;
-                if (DetailedWWR) {
-                    print(state.files.debug, "{},{:.2R},{:.2R}\n", thisZone.Name, thisZone.ExtGrossWallArea, thisZone.ExtWindowArea);
-                }
                 // Use AllSurfaceFirst which includes air boundaries
                 for (int SurfNum = thisSpace.AllSurfaceFirst; SurfNum <= thisSpace.AllSurfaceLast; ++SurfNum) {
                     auto &thisSurface = state.dataSurface->Surface(SurfNum);
 
                     if (thisSurface.Class == SurfaceClass::Roof) {
                         // Use Average Z for surface, more important for roofs than floors...
                         ++CeilCount;
-                        Z1 = minval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
-                        Z2 = maxval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
+                        Real64 Z1 = minval(thisSurface.Vertex, &Vector::z);
+                        Real64 Z2 = maxval(thisSurface.Vertex, &Vector::z);
                         //        ZCeilAvg=ZCeilAvg+(Z1+Z2)/2.d0
-                        ZCeilAvg += ((Z1 + Z2) / 2.0) * (thisSurface.GrossArea / ZoneCeilingArea(ZoneNum));
+                        ZCeilAvg += ((Z1 + Z2) / 2.0) * (thisSurface.GrossArea / thisZone.geometricCeilingArea);
                     }
                     if (thisSurface.Class == SurfaceClass::Floor) {
                         // Use Average Z for surface, more important for roofs than floors...
                         ++FloorCount;
-                        Z1 = minval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
-                        Z2 = maxval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
+                        Real64 Z1 = minval(thisSurface.Vertex, &Vector::z);
+                        Real64 Z2 = maxval(thisSurface.Vertex, &Vector::z);
                         //        ZFlrAvg=ZFlrAvg+(Z1+Z2)/2.d0
-                        ZFlrAvg += ((Z1 + Z2) / 2.0) * (thisSurface.Area / thisZone.FloorArea);
+                        ZFlrAvg += ((Z1 + Z2) / 2.0) * (thisSurface.GrossArea / thisZone.geometricFloorArea);
                     }
                     if (thisSurface.Class == SurfaceClass::Wall) {
                         // Use Wall calculation in case no roof & floor in zone
-                        ++Count;
-                        if (Count == 1) {
+                        ++WallCount;
+                        if (WallCount == 1) {
```

Comment:
Did you know you can do `if (++WallCount == 1)` and it would be the same thing?

Reply:
> Did you know you can do `if (++WallCount == 1)` and it would be the same thing?

No, I did not. I don't expect an `if` statement to change any values within it, so that's confusing to me. Since these are one-time calculations, I'm inclined to let it be.